docs: add note about typescript-eslint as peer dependency (#9820)
typescript-eslint/typescript-eslint: docs/developers/ESLint_Plugins.mdx

Those are all packages consumers are expected to be using already.

> The parser for TypeScript files generally requires `typescript-eslint`.
> However, you do not need to add the parser, and therefore `typescript-eslint`, as a peer dependency.
> Our setup guide ensures that the parser is automatically registered when configuring ESLint.

Reviewer: > The parser for TypeScript files generally requires `typescript-eslint`.

No, it's the other way around. There's no dependency on the `typescript-eslint` package in the parser.

> you do not need to add the parser ... as a peer dependency

This goes against what's stated earlier in this section - that `@typescript-eslint/parser` should be a peer dependency.

---

How about...

```suggestion
You don't need to declare any dependencies on `typescript-eslint` or `@typescript-eslint/eslint-plugin`.
Our setup guide ensures that the parser is automatically registered when configuring ESLint.
```
